### ISSUE TITLE
Odigos uninstall output is not helpful when odigos in not installed

### ISSUE THREAD

**Issue by blumamir** [keyval-dev/odigos#658]
When running the odigos cli `odigos uninstall`, there can be 2 options.

1. odigos is install, in which case the following output is printed to console which is **good**
```
➜  cli git:(odigos-deployment) odigos uninstall
About to uninstall Odigos from namespace odigos-system
Are you sure? [Y/n]: Y
Uninstalling Odigos Deployments                   ✔
Uninstalling Odigos DaemonSets                    ✔
Uninstalling Odigos ConfigMaps                    ✔
Uninstalling Odigos CRDs                          ✔
Uninstalling Odigos RBAC                          ✔
Uninstalling Odigos Secrets                       ✔
Uninstalling Namespace odigos-system              ✔
Waiting for namespace to be deleted               ✔
Rolling back odigos changes to pods
Rolling back odigos changes to namespaces
        ✔

SUCCESS: Odigos uninstalled.
```

2. If Odigos is not installed to begin with, the output is not very helpful:
```
➜  cli git:(odigos-deployment) odigos uninstall
About to uninstall Odigos from namespace odigos-system
Are you sure? [Y/n]: Y
Uninstalling Odigos Deployments                   ✔
Uninstalling Odigos DaemonSets                    ✔
Uninstalling Odigos ConfigMaps                    ✔
Uninstalling Odigos CRDs                          ✔
Uninstalling Odigos RBAC                          ✔
Uninstalling Odigos Secrets                       ✔
Uninstalling Namespace odigos-system              X
ERROR namespaces "odigos-system" not found
```

I think it is more useful to print something like "Odigos is not installed in the cluster" if the namespace is not found.

To check your fix, you can set up an odigos environment with [Quickstart](https://docs.odigos.io/intro). Then, make necessary changes in the `cli` module code, and run in terminal:
```sh
➜  odigos git:(main) cd cli
➜  cli git:(main) go run . uninstall
```

To run the cli uninstall command locally from code and verify your fix.

Thanks!

**Comment by HeetVekariya** [comment]
- @blumamir I understand the issue as to just change the print statement of the second case.
- If you find my solution correct, can you please assign me the issue.
- Also, are you using the older version, because i have already corrected the position of Tic marks(v) in this issue: https://github.com/keyval-dev/odigos/issues/614

**Comment by blumamir** [comment]
Yes I am using v0.1.75. 
Thanks for the help @HeetVekariya

**Comment by HeetVekariya** [comment]
@blumamir I am having the doubt that where exactly should i add the check for the namespace ?

```
func waitForNamespaceDeletion(ctx context.Context, client *kube.Client, ns string) {
	l := log.Print("Waiting for namespace to be deleted")
	wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
		_, err := client.CoreV1().Namespaces().Get(ctx, ns, metav1.GetOptions{})
		if err != nil {
			l.Success()
			return true, nil
		}
		return false, nil
	})
}
```


Or


```
l = log.Print("Rolling back odigos changes to namespaces")
		err = rollbackNamespaceChanges(ctx, client)
		if err != nil {
			l.Error(err)
		} else {
			l.Success()
		}
```

**Comment by blumamir** [comment]
@HeetVekariya 
Your suggested fix will still run and print this chunk of code:
```golang
		fmt.Printf("About to uninstall Odigos from namespace %s\n", ns)
		confirmed, err := confirm.Ask("Are you sure?")
		if err != nil || !confirmed {
			fmt.Println("Aborting uninstall")
			return
		}

		createKubeResourceWithLogging(ctx, "Uninstalling Odigos Deployments",
			client, cmd, ns, uninstallDeployments)
		createKubeResourceWithLogging(ctx, "Uninstalling Odigos DaemonSets",
			client, cmd, ns, uninstallDaemonSets)
		createKubeResourceWithLogging(ctx, "Uninstalling Odigos ConfigMaps",
			client, cmd, ns, uninstallConfigMaps)
		createKubeResourceWithLogging(ctx, "Uninstalling Odigos Services",
			client, cmd, ns, uninstallServices)
		createKubeResourceWithLogging(ctx, "Uninstalling Odigos CRDs",
			client, cmd, ns, uninstallCRDs)
		createKubeResourceWithLogging(ctx, "Uninstalling Odigos RBAC",
			client, cmd, ns, uninstallRBAC)
		createKubeResourceWithLogging(ctx, "Uninstalling Odigos Secrets",
			client, cmd, ns, uninstallSecrets)
		createKubeResourceWithLogging(ctx, fmt.Sprintf("Uninstalling Namespace %s", ns),
			client, cmd, ns, uninstallNamespace)
```

What I had in mind is to check if the [resources.GetOdigosNamespace function](https://github.com/keyval-dev/odigos/blob/c31ae9804bc49f1994cb248728912d660c33c204/cli/cmd/uninstall.go#L39) returns indication that no namespace is found, and if so, print a useful message and exit.

WDYT?

**Comment by HeetVekariya** [comment]
@blumamir so i think i need to change here

```
ns, err := resources.GetOdigosNamespace(client, ctx)
		if err != nil {
			ns = consts.DefaultNamespace
		}

```

**Comment by HeetVekariya** [comment]
@blumamir I am also thinking the same but not sure about that previously without talking to you so continued with other, but now can i check this:

```
errMessage := fmt.Errorf("Odigos is not installed in the cluster")
  ns, err := resources.GetOdigosNamespace(client, ctx)
  if err != nil {
	  if errors.IsNotFound(err) {
		  fmt.Println(errMessage)
		  os.Exit(1)
	  } else {
		  ns = consts.DefaultNamespace
	  }
  }
```

**Comment by HeetVekariya** [comment]
same as above function should i add for the ```odigos install``` issue, here https://github.com/keyval-dev/odigos/blob/main/cli/cmd/install.go in line 55 ?

**Comment by blumamir** [comment]
> @blumamir I am also thinking the same but not sure about that previously without talking to you so continued with other, but now can i check this:
> 
> ```
> errMessage := fmt.Errorf("Odigos is not installed in the cluster")
>   ns, err := resources.GetOdigosNamespace(client, ctx)
>   if err != nil {
> 	  if errors.IsNotFound(err) {
> 		  fmt.Println(errMessage)
> 		  os.Exit(1)
> 	  } else {
> 		  ns = consts.DefaultNamespace
> 	  }
>   }
> ```

This looks good to me. I wonder if we should use the default namespace if there is an `err`, as it probably indicates some issue in running a quest against k8s.
Should we maybe `os.Exit` on any error? but still format a specific message for the first case where a ns is not found vs some general err like network.

Please feel free to open a PR with a fix and we can discuss there 🙏

**Comment by blumamir** [comment]
> here https://github.com/keyval-dev/odigos/blob/main/cli/cmd/install.go in line 55 ?

You are doing awesome work, thanks! if you want to learn something new for using github, I have 2 tips:

1. if you want to get a link for a specific line in a specific file, you can click on that file number (`55`) in your case, and choose `Cope permalink`

![image](https://github.com/keyval-dev/odigos/assets/22870745/d32c2008-70be-43a7-b4b0-6ef6fd340ac0)

This link will still point to the right line even if the source file is changed in the future.

2. instead of pasting urls in github comments, you can use [Markdown links](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#links), so you can write cleaner text which is easier to read:

[here](https://github.com/keyval-dev/odigos/blob/5dd675dc6d9735eb037800bf6759c3d0a9f79f1a/cli/cmd/install.go#L55) - This link is written like this:

![image](https://github.com/keyval-dev/odigos/assets/22870745/c6629ab6-8200-4df4-b01f-9734754d1534)

**Comment by HeetVekariya** [comment]
> > here https://github.com/keyval-dev/odigos/blob/main/cli/cmd/install.go in line 55 ?
> 
> You are doing awesome work, thanks! if you want to learn something new for using github, I have 2 tips:
> 
> 1. if you want to get a link for a specific line in a specific file, you can click on that file number (`55`) in your case, and choose `Cope permalink`
> 
> ![image](https://user-images.githubusercontent.com/22870745/276341303-d32c2008-70be-43a7-b4b0-6ef6fd340ac0.png)
> 
> This link will still point to the right line even if the source file is changed in the future.
> 
> 2. instead of pasting urls in github comments, you can use [Markdown links](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#links), so you can write cleaner text which is easier to read:
> 
> [here](https://github.com/keyval-dev/odigos/blob/5dd675dc6d9735eb037800bf6759c3d0a9f79f1a/cli/cmd/install.go#L55) - This link is written like this:
> 
> ![image](https://user-images.githubusercontent.com/22870745/276342248-c6629ab6-8200-4df4-b01f-9734754d1534.png)

- I want to reply like that but i don't how to do it
- Thank you for the tip :+1:

**Comment by HeetVekariya** [comment]
@blumamir I am facing error due to recent changes in this:
https://github.com/keyval-dev/odigos/blob/3319cab6c2ea6034d3aa25ca073a1dd8602339a3/instrumentor/go.mod#L4

This generates error:
![image](https://github.com/keyval-dev/odigos/assets/91054457/65ca84c4-4dee-4b2c-81c1-af866e2aee58)

Do i need to update my go lang version ? 
My current version is: ```go version go1.18.1 linux/amd64```

**Comment by HeetVekariya** [comment]
> @blumamir I am facing error due to recent changes in this:
> 
> https://github.com/keyval-dev/odigos/blob/3319cab6c2ea6034d3aa25ca073a1dd8602339a3/instrumentor/go.mod#L4
> 
> This generates error: ![image](https://user-images.githubusercontent.com/91054457/276574514-65ca84c4-4dee-4b2c-81c1-af866e2aee58.png)
> 
> Do i need to update my go lang version ? My current version is: `go version go1.18.1 linux/amd64`

SOLVED!!
Just needed to update the go lang version

**Comment by HeetVekariya** [comment]
close this as completed

**Comment by HeetVekariya** [comment]
fixed in #673

**Comment by HeetVekariya** [comment]
fixes #658
